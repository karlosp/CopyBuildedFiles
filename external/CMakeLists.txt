cmake_minimum_required(VERSION 3.1)
add_subdirectory(json)
add_subdirectory(spdlog)
add_subdirectory(expected)
add_subdirectory(ConvertProjectsFromBuildAll)
