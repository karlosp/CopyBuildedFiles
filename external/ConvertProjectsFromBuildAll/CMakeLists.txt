cmake_minimum_required(VERSION 3.11)
project(ConvertProjectsFromBuildAll
        VERSION 0.0.1
        LANGUAGES CXX
        DESCRIPTION "ConvertProjectsFromBuildAll")

set(CMAKE_CXX_STANDARD 17)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
        

#-------------------
# Add *.hpp and *.cpp files to project Visual Studio ${PROJECT_NAME}.vcxproj
#-------------------
set(PSD ${PROJECT_SOURCE_DIR})
set(HEADER_FILES  ${PSD}/ConvertProjectsFromBuildAll.hpp
                  )
                  
set(SOURCE_FILES  ${PSD}/ConvertProjectsFromBuildAll.cpp
                  ${PSD}/main.cpp 
                  )

add_executable(${PROJECT_NAME} ${SOURCE_FILES} ${HEADER_FILES})
#add_subdirectory(../spdlog ./spdlog2)

target_include_directories(${PROJECT_NAME} PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
target_link_libraries(${PROJECT_NAME} spdlog)

#### Visual Studio stuff #####
# Set as startup project
set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT ${PROJECT_NAME})
