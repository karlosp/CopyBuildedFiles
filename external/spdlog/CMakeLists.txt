cmake_minimum_required(VERSION 3.1)
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)

#---------------------------------------------------------------------------------------
# spdlog target
#---------------------------------------------------------------------------------------
add_library(spdlog INTERFACE)
add_library(spdlog::spdlog ALIAS spdlog)

# Check if spdlog is being used directly or via add_subdirectory
set(SPDLOG_MASTER_PROJECT OFF)
if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
  set(SPDLOG_MASTER_PROJECT ON)
endif()

target_include_directories(
    spdlog
    INTERFACE
    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
    "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>"
)

set(HEADER_BASE "${CMAKE_CURRENT_SOURCE_DIR}/include")

# register project in CMake user registry
export(PACKAGE ${PROJECT_NAME})

file(GLOB_RECURSE spdlog_include_SRCS "${HEADER_BASE}/*.h")
add_custom_target(spdlog_headers_for_ide SOURCES ${spdlog_include_SRCS})