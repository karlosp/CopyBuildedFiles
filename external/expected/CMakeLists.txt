#############################################################################
# CMake stuff                                                               # 
#############################################################################
cmake_minimum_required(VERSION 3.10.0)
project(expected CXX)

add_library(${PROJECT_NAME} INTERFACE)
target_include_directories(${PROJECT_NAME} INTERFACE include/)